File: ClientApp/src/app/models/ticket.ts
export interface Ticket {
  id?: number;
  ticketId?: number;
  unitId: number;
  tenantId: number;
  propertyId?: number;
  title: string;
  description: string;
  category?: string; // 'Plumbing' | 'Electrical' | 'HVAC' | etc.
  priority: string; // 'Low' | 'Medium' | 'High' | 'Emergency'
  status: string; // 'Open' | 'In Progress' | 'Resolved' | 'Closed' | 'Denied'
  createdAt?: string | Date;
  updatedAt?: string | Date;
  resolvedAt?: string | Date;
  unit?: any;
  tenant?: any;
  tenantName?: string;
  comments?: TicketComment[];
}

export interface TicketComment {
  id?: number;
  commentId?: number;
  ticketId: number;
  userId: number;
  comment: string;
  createdAt?: string | Date;
  user?: any;
}

File: Server/Models/Ticket.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Server.Models
{
    public class Ticket
    {
        [Key]
        public int TicketId { get; set; }
        
        [Required]
        public int UnitId { get; set; }
        
        [Required]
        public int TenantId { get; set; }
        
        [Required]
        [StringLength(100)]
        public string Title { get; set; } = string.Empty;
        
        [Required]
        public string Description { get; set; } = string.Empty;
        
        [Required]
        [StringLength(20)]
        public string Priority { get; set; } = string.Empty; // Low, Medium, High, Emergency
        
        [Required]
        [StringLength(20)]
        public string Status { get; set; } = "New"; // New, Assigned, InProgress, OnHold, Resolved, Closed, Denied
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Navigation properties
        [ForeignKey("UnitId")]
        public Unit? Unit { get; set; }
        
        [ForeignKey("TenantId")]
        public User? Tenant { get; set; }
        
        public ICollection<TicketComment>? Comments { get; set; }
    }
}
